src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -24,15 +72,50 @@ class BigqueryDataProvider(DatabaseDataProvider):
     illegal_column_name_chars_replacement: list[str] = ["_"]
 
     _create_schema_statement: str = "CREATE SCHEMA IF NOT EXISTS {} OPTIONS (location='{}')"
+    NATIVE_PATHS = {
+        FileLocation.GS: "load_gs_file_to_table",
+        FileLocation.S3: "load_s3_file_to_table",
+        FileLocation.LOCAL: "load_local_file_to_table",
+    }
+    NATIVE_AUTODETECT_SCHEMA_CONFIG: Mapping[FileLocation, Mapping[str, list[FileType] | Callable]] = {
+        FileLocation.GS: {
+            "filetype": [FileType.CSV, FileType.NDJSON, FileType.PARQUET],
+            "method": lambda table, file: None,
+        },
+    }
+
+    FILE_PATTERN_BASED_AUTODETECT_SCHEMA_SUPPORTED: set[FileLocation] = {
+        FileLocation.GS,
+        FileLocation.LOCAL,
+    }
+
+    NATIVE_LOAD_EXCEPTIONS: Any = (
+        GoogleNotFound,
+        ClientError,
+        GoogleAPIError,
+        RetryError,
+        InvalidArgument,
+        Unauthorized,
+        Forbidden,
+        Conflict,
+        TooManyRequests,
+        ResourceExhausted,
+        ServerError,
+        Unknown,
+        ServiceUnavailable,
+        InvalidResponse,
+        DatabaseCustomError,
+    )
+    OPTIONS_CLASS = BigqueryOptions
 
     def __init__(
         self,
         dataset: Table,
         transfer_mode,
-        transfer_params: TransferIntegrationOptions = TransferIntegrationOptions(),
+        transfer_params: BigqueryOptions = BigqueryOptions(),
```

Comment:
```suggestion
        transfer_params: BigqueryOptions = OPTIONS_CLASS(),
```
Can this be like this?

Reply:
No, it has to be `BigqueryOptions()` class here as it is what we need to validate